### PR TITLE
Consistent log messages

### PR DESCRIPTION
**Pull request by adams85**
### Describe the purpose of your pull request

Revises log messages to make them consistent with other SDKs and also adds event ids for easier log event identification.

Also, partially implements structured logging as the [logging abstraction](https://docs.python.org/3/library/logging.html) makes this possible. (Argument names are not passed at the moment though.)

### Requirement checklist (only if applicable)

- [x] I have covered the applied changes with automated tests.
- [x] I have executed the full automated test set against my changes.
- [x] I have validated my changes against all supported platform versions.
- [x] I have read and accepted the [contribution agreement](https://github.com/configcat/legal/blob/main/contribution-agreement.md).

### CODE REVIEW

File: configcatclient/rolloutevaluator.py [hunk 1]
```diff
@@ -39,31 +39,31 @@ def evaluate(self, key, user, default_value, default_variation_id, settings):
...
-                    'Returning default_value: [{}]. ' \
-                    'Here are the available keys: {}'.format(key, key, str(default_value),
-                                                             ', '.join(list(settings)))
-
-            self.log.error(error)
-            return default_value, default_variation_id, None, None, error
+            error = 'Failed to evaluate setting \'%s\' (the key was not found in config JSON). ' \
+                    'Returning the `%s` parameter that you specified in your application: \'%s\'. ' \
+                    'Available keys: %s.'
+            errorArgs = (key, 'default_value', str(default_value), ', '.join(list(settings)))
```
Comment (adams85) [inline configcatclient/rolloutevaluator.py:45]: Available keys should be surrounded with single quotes like `'Key1', 'Key2', 'Key3'`

File: configcatclient/logger.py [hunk 2]
```diff
@@ -1,26 +1,31 @@
...
 
     def exception(self, msg, *args, **kwargs):
-        self.error(msg, *args, **kwargs)
+        self._hooks.invoke_on_error(Logger.format(msg, args, sys.exc_info()[1]))
+        super().exception(msg, *args, **kwargs)
+
+    @staticmethod
+    def format(msg, args, exc = None):
+        msg = msg % args if len(args) > 0 else msg
+        return msg if exc is None else msg + "\n" + str(exc)
```
Comment (kp-cat) [inline configcatclient/logger.py:31]:
```suggestion
        return msg if exc is None else msg + '\n' + str(exc)
```

Comment (kp-cat) [reply configcatclient/logger.py:31]: Could you please change the format of `event_id = 1000` to `event_id=1000` (remove the extra spaces)

File: configcatclient/configcatclient.py [hunk 3]
```diff
@@ -242,7 +243,7 @@ def get_key_and_value(self, variation_id):
                 if variation_id == rollout_percentage_item.get(VARIATION_ID):
                     return KeyValue(key, rollout_percentage_item[VALUE])
 
-        self.log.error('Could not find the setting for the given variation_id: ' + variation_id)
+        self.log.error('Could not find the setting for the specified variation ID: \'%s\'.', variation_id, event_id = 2011)
```
Comment (kp-cat) [inline configcatclient/configcatclient.py:246]: I would use `variation_id` here in the log message on python:
```suggestion
        self.log.error('Could not find the setting for the specified variation_id: \'%s\'.', variation_id, event_id = 2011)
```

File: configcatclient/configcatclient.py [hunk 4]
```diff
@@ -37,8 +38,10 @@ def get(cls, sdk_key, options=None):
             client = cls._instances.get(sdk_key)
             if client is not None:
                 if options is not None:
-                    client.log.warning('Client for sdk_key `%s` is already created and will be reused; '
-                                       'options passed are being ignored.' % sdk_key)
+                    client.log.warning('There is an existing client instance for the specified SDK Key. ' \
+                                       'No new client instance will be created and the specified options are ignored. ' \
```
Comment (kp-cat) [inline configcatclient/configcatclient.py:42]: remove the unnecessary blackslashes

```suggestion
                    client.log.warning('There is an existing client instance for the specified SDK Key. '
                                       'No new client instance will be created and the specified options are ignored. '
```

File: configcatclient/configcatclient.py [hunk 5]
```diff
@@ -139,10 +142,10 @@ def get_value_details(self, key, default_value, user=None):
         """
         settings, fetch_time = self.__get_settings()
         if settings is None:
-            message = 'Evaluating get_value_details(\'{}\') failed. Cache is empty. ' \
-                      'Returning default_value in your get_value_details call: [{}].'.format(key, str(default_value))
-            self.log.error(message)
-            details = EvaluationDetails.from_error(key, default_value, message)
+            message = 'Config JSON is not present. Returning the `%s` parameter that you specified in your application: \'%s\'.'
```
Comment (kp-cat) [inline configcatclient/configcatclient.py:145]: Compare to the previous version here we lost info: Evaluating get_value_details(. is it acceptable?

File: configcatclient/configcatclient.py [hunk 6]
```diff
@@ -318,8 +318,8 @@ def set_online(self):
         """
         if self._config_service:
             self._config_service.set_online()
-
-        self.log.debug('Switched to ONLINE mode.')
+        else:
+            self.log.warning('Client is configured to use the `%s` override behavior, thus `%s()` has no effect.', 'LocalOnly', 'set_online', event_id=3202)
```
Comment (kp-cat) [inline configcatclient/configcatclient.py:322]: Why are we using %s here instead of putting the set_online directly into the string literal?

**Review by kp-cat — APPROVED** [review]

**Pull request merged**